test/unit/org/apache/cassandra/cql3/PasswordObfuscatorTest.java

```diff
@@ -0,0 +1,162 @@
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD '123'"));
+    }
+
+    @Test
+    public void testCreateUserWithoutPassword()
+    {
+        assertEquals("CREATE USER user1", obfuscator.obfuscate("CREATE USER user1"));
+    }
+
+    @Test
+    public void testAlterUserWithPassword()
+    {
+        assertEquals(format("ALTER USER user1 with PASSWORD%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with PASSWORD '123'"));
+    }
+
+    @Test
+    public void testAlterUserWithPasswordMixedCase()
+    {
+        assertEquals(format("ALTER USER user1 with paSSwoRd%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with paSSwoRd '123'"));
+    }
+
+    @Test
+    public void testAlterUserWithPasswordWithNewLine()
+    {
+        assertEquals(format("ALTER USER user1 with PASSWORD%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("ALTER USER user1 with PASSWORD\n'123'"));
+    }
+
+    @Test
+    public void testPasswordWithNewLinesObfuscation()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD 'a\nb'"));
+    }
+
+    @Test
+    public void testEmptyPasswordObfuscation()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD ''"));
+    }
+
+    @Test
+    public void testPasswordWithSpaces()
+    {
+        assertEquals(String.format("CREATE USER user1 with PASSWORD%s", PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("CREATE USER user1 with PASSWORD 'p a ss wor d'"));
+    }
+
+    @Test
+    public void testSimpleBatch()
+    {
+        assertEquals(format("BEGIN BATCH \n" +
+                            "    CREATE ROLE alice1 WITH PASSWORD%s",
+                            PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("BEGIN BATCH \n" +
+                                          "    CREATE ROLE alice1 WITH PASSWORD = 'alice123' and LOGIN = true; \n" +
+                                          "APPLY BATCH;"));
+    }
+
+    @Test
+    public void testComplexBatch()
+    {
+        assertEquals(format("BEGIN BATCH \n" +
+                            "    CREATE ROLE alice1 WITH PASSWORD%s",
+                            PasswordObfuscator.OBFUSCATION_TOKEN),
+                     obfuscator.obfuscate("BEGIN BATCH \n" +
+                                          "    CREATE ROLE alice1 WITH PASSWORD = 'alice123' and LOGIN = true; \n" +
+                                          "    CREATE ROLE alice2 WITH PASSWORD = 'alice123' and LOGIN = true; \n" +
+                                          "APPLY BATCH;"));
+    }
+}
```

Comment:
Is it possible to add a test case covering an invalid DCL statement, something that you identified in review? if it is already added, ignore my comment.

`INFO  [Native-Transport-Requests-1] 2021-06-09 23:38:05,965 FileAuditLogger.java:51 - user:anonymous|host:localhost/127.0.0.1:7000|source:/127.0.0.1|port:63360|timestamp:1623296285965|type:REQUEST_FAILURE|category:ERROR|operation:BEGIN BATCH CREATE ROLE alice WITH PASSWORD = 'password_a' AND LOGIN = true; CREATE ROLE new_role; APPLY BATCH;; line 1:12 mismatched input 'CREATE' expecting K_APPLY (BEGIN BATCH [CREATE]...)
`